databricks/sdk/mixins/sql.py

```diff
@@ -0,0 +1,231 @@
+                          catalog: Optional[str] = None,
+                          schema: Optional[str] = None,
+                          timeout: timedelta = timedelta(minutes=20),
+                          ) -> Iterator[Row]:
+        """(Experimental) Execute a query and iterate over all available records.
+
+        This method internally calls `execute()` to handle async execution and timeouts.
+
+        Applications, that need to a more traditional SQL Python APIs with cursors and low result fetching latency,
+        should use Databricks SQL connector for Python.
+
+        :param warehouse_id: str
+          Warehouse upon which to execute a statement.
+        :param statement: str
+          SQL statement to execute
+        :param byte_limit: int (optional)
+          Applies the given byte limit to the statement's result size. Byte counts are based on internal
+          representations and may not match measurable sizes in the JSON format.
+        :param catalog: str (optional)
+          Sets default catalog for statement execution, similar to `USE CATALOG` in SQL.
+        :param schema: str (optional)
+          Sets default schema for statement execution, similar to `USE SCHEMA` in SQL.
+        :param timeout: timedelta (optional)
+          Timeout after which the query is cancelled. If timeout is less than 50 seconds,
+          it is handled on the server side. If the timeout is greater than 50 seconds,
+          Databricks SDK for Python cancels the statement execution and throws `TimeoutError`.
+        :return: Iterator[Row]
+        """
+        execute_response = self.execute(warehouse_id,
+                                        statement,
+                                        byte_limit=byte_limit,
+                                        catalog=catalog,
+                                        schema=schema,
+                                        timeout=timeout)
+        col_names = []
+        col_conv = []
+        for col in execute_response.manifest.schema.columns:
+            col_names.append(col.name)
+            conv = self.type_converters.get(col.type_name, None)
+            if conv is None:
+                msg = f"{col.name} has no {col.type_name.value} converter"
+                raise ValueError(msg)
+            col_conv.append(conv)
+        row_factory = type("Row", (Row, ), {"__columns__": col_names})
+        result_data = execute_response.result
+        while True:
+            for data in result_data.data_array:
+                # enumerate() + iterator + tuple constructor makes it more performant
+                # on larger humber of records for Python, even though it's less
+                # readable code.
+                yield row_factory(col_conv[i](value) for i, value in enumerate(data))
+            if result_data.next_chunk_index is None:
+                return
+            # TODO: replace once ES-828324 is fixed
```

Comment:
This is almost fixed. But I'm curious whether we want to refer to internal ES tickets in our public repos?

Reply:
oops... but ... ES tickets _could be referred_ outside with caution. let's start a thread :)

Reply:
Fix is rolling out as we speak should hit all regions before end of next week